editor/src/components/canvas/canvas-strategies/post-action-options/post-action-paste.ts

```diff
@@ -349,13 +419,45 @@ export const PropsReplacedPastePostActionChoice = (
           elementPasteWithMetadata: dataWithPropsReplaces,
           targetOriginalPathTrees: postActionMenuData.targetOriginalPathTrees,
           canvasViewportCenter: postActionMenuData.canvasViewportCenter,
+          originalAllElementProps: postActionMenuData.originalAllElementProps,
           reparentStrategy: null,
           insertionPosition: null,
         },
       ),
   }
 }
 
+export function getUidsFromJSXElementChild(element: JSXElementChild): string[] {
+  switch (element.type) {
+    case 'JSX_ELEMENT':
+    case 'JSX_FRAGMENT':
+      return [element.uid, ...element.children.flatMap(getUidsFromJSXElementChild)]
+    case 'JSX_CONDITIONAL_EXPRESSION':
+      return [
+        element.uid,
+        ...getUidsFromJSXElementChild(element.whenTrue),
+        ...getUidsFromJSXElementChild(element.whenFalse),
+      ]
+    case 'ATTRIBUTE_FUNCTION_CALL':
+    case 'ATTRIBUTE_NESTED_ARRAY':
+    case 'ATTRIBUTE_NESTED_OBJECT':
+    case 'ATTRIBUTE_OTHER_JAVASCRIPT':
+    case 'ATTRIBUTE_VALUE':
+    case 'JSX_TEXT_BLOCK':
+    case 'JSX_MAP_EXPRESSION':
+      return [element.uid]
+    default:
+      assertNever(element)
+  }
+}
+
+function zip<A, B, C>(one: A[], other: B[], make: (a: A, b: B) => C): C[] {
+  const doZip = (oneInner: A[], otherInner: B[]) =>
+    oneInner.map((elem, idx) => make(elem, otherInner[idx]))
+
+  return one.length < other.length ? doZip(one, other) : doZip(one.slice(0, other.length), other)
+}
```

Comment:
This looks general enough to go to array-utils

Reply:
moved it in [b0603b5](https://github.com/concrete-utopia/utopia/pull/3890/commits/b0603b5a3104002e93c99db0c4db49297f2bb97a)